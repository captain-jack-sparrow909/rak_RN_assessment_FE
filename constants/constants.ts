export const BACKEND_URL = 'http://localhost:8080/users';

export const POST = 'POST';