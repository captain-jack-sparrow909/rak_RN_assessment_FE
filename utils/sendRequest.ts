import axios from "axios";

export const sendRequest = async (url: string, method: string, data: any) => {
  try {
    const response = await axios({
      url,
      method,
      data,
    });
    return response.data;
  } catch (error) {
    console.error(error);
  }
};
