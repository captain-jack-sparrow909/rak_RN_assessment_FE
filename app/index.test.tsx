import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react-native';
import RegisterScreen from './index';
import { Alert } from 'react-native';
import { sendRequest } from '@/utils/sendRequest';
import { BACKEND_URL, POST } from '@/constants/constants';

jest.mock('@/utils/sendRequest');
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('RegisterScreen', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<RegisterScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should update form fields correctly', async () => {
    const { getByTestId, toJSON } = render(<RegisterScreen />);
    const nameInput = getByTestId('fullName');
    const emailInput = getByTestId('email');
    const passwordInput = getByTestId('password');

    fireEvent.changeText(nameInput, 'John Doe');
    fireEvent.changeText(emailInput, 'john@example.com');
    fireEvent.changeText(passwordInput, 'password123');
    
    expect(toJSON()).toMatchSnapshot();
  });

  it('should show an alert when the form is incomplete', () => {
    const alertSpy = jest.spyOn(Alert, 'alert');
    const { getByText } = render(<RegisterScreen />);
    const signUpButton = getByText('SIGN UP');

    fireEvent.press(signUpButton);

    expect(alertSpy).toHaveBeenCalledWith(
      'Incomplete Form',
      'Please fill in all fields',
      [{ text: 'OK' }]
    );
  });

  it('should send a request when the form is complete', async () => {
    const { getByTestId, getByText } = render(<RegisterScreen />);
    const nameInput = getByTestId('fullName');
    const emailInput = getByTestId('email');
    const passwordInput = getByTestId('password');
    const signUpButton = getByTestId('signUp');
    const checkbox = getByTestId('checkbox');


        fireEvent.changeText(nameInput, 'John Doe');
        fireEvent.changeText(emailInput, 'john@example.com');
        fireEvent.changeText(passwordInput, 'password123');
        fireEvent.press(checkbox);
        fireEvent.press(signUpButton);
  
    await waitFor(() => {
      expect(sendRequest).toHaveBeenCalledWith(
        `${BACKEND_URL}`,
        POST,
        {
          name: 'John Doe',
          email: 'john@example.com',
          password: 'password123',
        }
      );
    });
  });
});