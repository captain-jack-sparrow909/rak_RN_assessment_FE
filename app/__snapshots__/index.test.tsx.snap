// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={1}
      style={
        {
          "height": 220,
          "width": 220,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
        "textAlign": "left",
      }
    }
  >
    Create Account
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text />
    <TextInput
      onChangeText={[Function]}
      placeholder="Full Name"
      style={
        {
          "flex": 1,
          "fontSize": 16,
        }
      }
      testID="fullName"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text />
    <TextInput
      onChangeText={[Function]}
      placeholder="Email address"
      style={
        {
          "flex": 1,
          "fontSize": 16,
        }
      }
      testID="email"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        {
          "flex": 1,
          "fontSize": 16,
        }
      }
      testID="password"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "marginVertical": 10,
        "paddingLeft": 4,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 8,
              "width": 20,
            },
            false,
          ]
        }
      />
    </View>
    <Text
      style={
        {
          "fontSize": 14,
          "marginLeft": 8,
        }
      }
    >
      I agree with 
      <Text
        style={
          {
            "color": "#E63946",
            "textDecorationLine": "underline",
          }
        }
      >
        Terms
      </Text>
       and 
      <Text
        style={
          {
            "color": "#E63946",
            "textDecorationLine": "underline",
          }
        }
      >
        Privacy
      </Text>
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E63946",
        "borderRadius": 8,
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "justifyContent": "center",
        "marginVertical": 10,
        "minWidth": 380,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
    testID="signUp"
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      SIGN UP
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "#000",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "justifyContent": "center",
        "marginVertical": 10,
        "minWidth": 380,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
    testID="googleSignUp"
  >
    <Text>
      <Text />
    </Text>
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          {
            "color": "#282828",
            "fontWeight": "regular",
          },
        ]
      }
    >
      Sign Up with Google
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 14,
          "marginTop": 20,
        }
      }
    >
      Already have an account? 
      <Text
        style={
          {
            "color": "#E63946",
            "textDecorationLine": "underline",
          }
        }
      >
        Log In
      </Text>
    </Text>
  </View>
</View>
`;

exports[`RegisterScreen should update form fields correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={1}
      style={
        {
          "height": 220,
          "width": 220,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
        "textAlign": "left",
      }
    }
  >
    Create Account
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#a9a9a9",
            "fontSize": 24,
          },
          {
            "marginRight": 10,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î¶­
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Full Name"
      style={
        {
          "flex": 1,
          "fontSize": 16,
        }
      }
      testID="fullName"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#a9a9a9",
            "fontSize": 24,
          },
          {
            "marginRight": 10,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î´µ
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Email address"
      style={
        {
          "flex": 1,
          "fontSize": 16,
        }
      }
      testID="email"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#a9a9a9",
            "fontSize": 24,
          },
          {
            "marginRight": 10,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î³
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        {
          "flex": 1,
          "fontSize": 16,
        }
      }
      testID="password"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "marginVertical": 10,
        "paddingLeft": 4,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 8,
              "width": 20,
            },
            false,
          ]
        }
      />
    </View>
    <Text
      style={
        {
          "fontSize": 14,
          "marginLeft": 8,
        }
      }
    >
      I agree with 
      <Text
        style={
          {
            "color": "#E63946",
            "textDecorationLine": "underline",
          }
        }
      >
        Terms
      </Text>
       and 
      <Text
        style={
          {
            "color": "#E63946",
            "textDecorationLine": "underline",
          }
        }
      >
        Privacy
      </Text>
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E63946",
        "borderRadius": 8,
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "justifyContent": "center",
        "marginVertical": 10,
        "minWidth": 380,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
    testID="signUp"
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      SIGN UP
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "#000",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "justifyContent": "center",
        "marginVertical": 10,
        "minWidth": 380,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
    testID="googleSignUp"
  >
    <Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î³¶
      </Text>
    </Text>
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          {
            "color": "#282828",
            "fontWeight": "regular",
          },
        ]
      }
    >
      Sign Up with Google
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 14,
          "marginTop": 20,
        }
      }
    >
      Already have an account? 
      <Text
        style={
          {
            "color": "#E63946",
            "textDecorationLine": "underline",
          }
        }
      >
        Log In
      </Text>
    </Text>
  </View>
</View>
`;
